client/executor/src/native_executor.rs

```diff
@@ -269,12 +290,109 @@ impl<D: NativeExecutionDispatch> RuntimeInfo for NativeExecutor<D> {
 			runtime_code,
 			ext,
 			false,
-			|_instance, version, _ext|
+			|_module, _instance, version, _ext|
 				Ok(version.cloned().ok_or_else(|| Error::ApiError("Unknown version".into()))),
 		)
 	}
 }
 
+pub struct RuntimeInstanceSpawn {
+	module: Arc<dyn WasmModule>,
+	forks: parking_lot::Mutex<HashMap<u64, mpsc::Receiver<Vec<u8>>>>,
+	counter: AtomicU64,
+	scheduler: Box<dyn sp_core::traits::SpawnNamed>,
+}
+
+impl sp_io::RuntimeSpawn for RuntimeInstanceSpawn {
+	fn dyn_dispatch(&self, dispatcher_ref: u32, func: u32, data: Vec<u8>) -> u64 {
+		let new_handle = self.counter.fetch_add(1, Ordering::Relaxed);
+
+		let (sender, receiver) = mpsc::channel();
```

Comment:
with `mpsc::channel(1)` you mean https://doc.rust-lang.org/std/sync/mpsc/fn.sync_channel.html ?

We don't need any bounding, since we only ever send 1 message

and std lacks `oneshot` channel afaik